From 6ef6d4154aecd88658446233ddbad970830d110e Mon Sep 17 00:00:00 2001
From: Laxminath Kasam <lkasam@codeaurora.org>
Date: Fri, 16 Oct 2020 17:53:22 +0530
Subject: [PATCH] asoc: msm-pcm-q6-v2: Update memset for period size

tinycap test can attempt with different size to
read from driver and need to avoid access more
than period size.

Change-Id: Ifa4ddfb086bd83aa981da62e88da3a9395f5aabc
Signed-off-by: Laxminath Kasam <lkasam@codeaurora.org>
---
 techpack/audio/asoc/msm-pcm-q6-v2.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/techpack/audio/asoc/msm-pcm-q6-v2.c b/techpack/audio/asoc/msm-pcm-q6-v2.c
index ebcd54854513..edbe943ebb31 100755
--- a/techpack/audio/asoc/msm-pcm-q6-v2.c
+++ b/techpack/audio/asoc/msm-pcm-q6-v2.c
@@ -953,9 +953,9 @@ static int msm_pcm_capture_copy(struct snd_pcm_substream *substream,
 			goto fail;
 		}
 
-		if (size == 0 || size < fbytes) {
-			memset(bufptr + offset + size, 0, fbytes - size);
-			size = xfer = fbytes;
+		if (size == 0 || size < prtd->pcm_count) {
+			memset(bufptr + offset + size, 0, prtd->pcm_count - size);
+			size = xfer = prtd->pcm_count;
 		}
 
 		if (copy_to_user(buf, bufptr+offset, xfer)) {
